### DISCUSSION TITLE
Future features discussion: Additional/alternative state backends

### DISCUSSION THREAD

**Discussion by aaronsteers** [meltano/meltano#6952] [Ideas]
Drop a comment here if you have suggestions.

Related to:

- #6911

**Comment by pnadolny13** [comment]
I've heard a few people in slack ask about a snowflake state backend to simplify infrastructure if they dont want a standalone OLTP db or S3/GCP/Azure bucket and already need Snowflake.

- https://meltano.slack.com/archives/C01UTUSP34M/p1661967818591489
- https://meltano.slack.com/archives/C01TCRBBJD7/p1649967936914359

I'd also consider using this. It would allow us to avoid creating an S3 bucket for Squared and it would also give me query access to state files. Not sure what use cases the query access unlocks but im sure it unlocks something.

**Reply by tayloramurphy** [reply]
@pnadolny13 snowflake makes me think about their JSON size limit of 16 MB. Do other databases have a similar limits?
